core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -48,12 +48,19 @@ void shouldCheckIfTheUnreleasedCommitsExist() throws IOException {
         when(plugin.getDetails()).thenReturn(Map.of(
             SCMLinkValidationProbe.KEY, ProbeResult.success("scm", "The plugin SCM link is valid"))
         );
-        when(plugin.getScm()).thenReturn("https://github.com/jenkinsci/git-plugin/");
+        when(plugin.getScm()).thenReturn("https://github.com/jenkinsci/git-plugin/tree/master/src");
         when(ctx.getScmRepository()).thenReturn(Files.createTempDirectory(UUID.randomUUID().toString()));
-        final ProbeResult r = probe.apply(plugin, ctx);
+        final ProbeResult probeResult = probe.apply(plugin, ctx);
 
-        assertThat(r.id()).isEqualTo("last-commit-date");
-        assertThat(r.status()).isEqualTo(ResultStatus.SUCCESS);
+        assertThat(probeResult.id()).isEqualTo("unreleased-production-changes");
+        assertThat(probeResult.status()).isEqualTo(ResultStatus.SUCCESS);
```

Comment:
How do I test for the scenario where "unreleased commits exits"

Reply:
Hey @alecharp 

Thank you for explaining this in detail. I have implemented first few steps.